### ISSUE TITLE
【记录】PR-CE-Framework 流水线 PIR 单测修复

### ISSUE THREAD

**Issue by ooooo-create** [PaddlePaddle/Paddle#65943]
# PR-CE-Framework 流水线 PIR 单测修复

在 https://github.com/PaddlePaddle/Paddle/pull/65893 中开启 FLAGS_enable_pir_api = True，得到需要适配 PIR 的单测列表

## 方法

1. PR-CE-Framework 对应的测试在 PaddleTest 目录中，https://github.com/PaddlePaddle/PaddleTest/tree/develop/framework/api ， 使用 pytest 进行单元测试

2. 测试命令
```
FLAGS_enable_pir_api=1 python -m pytest test_functional_utils_weight_norm.py
FLAGS_enable_pir_api=1 python -m pytest test_functional_utils_weight_norm.py::test_weight_norm0
```

3. 

## 记录

| 单测名称 | 初步问题（待）  | 解决 |
| --- | --- | --- |
| test_vjp.py | instance Variable   | ✅[PaddleTest#2881](https://github.com/PaddlePaddle/PaddleTest/pull/2881)  |
| test_Jacobian.py | 缺少 Value 的判断逻辑 |  ✅[PaddleTest#2881](https://github.com/PaddlePaddle/PaddleTest/pull/2881)  |
|                   |             |
| test_allclose.py | 测试用例4，PIR下的结果应该是正确的，之前静态图的bug |  ✅[PaddleTest#2885](https://github.com/PaddlePaddle/PaddleTest/pull/2885) |
| test_kthvalue.py | 期望结果设置错误,属于静态图bug | ✅[PaddleTest#2885](https://github.com/PaddlePaddle/PaddleTest/pull/2885) |
| test_squeeze.py |  paddle.static.gradients 对于无关输出的单个input梯度，返回的是 [None]，旧静态图下是 [],传入到 fetch_list 当中 |  ✅[PaddleTest#2883](https://github.com/PaddlePaddle/PaddleTest/pull/2883)  |
| test_mean.py | 出现 `Windows fatal exception: access violation` | ✅[PaddleTest#2885](https://github.com/PaddlePaddle/PaddleTest/pull/2885)  |
| |   |
| test_TransformerDecoder.py | PIR 下Tensor的shape是[]，旧IR下是()，添加PIRmode判断即可 | ✅[PaddleTest#2877](https://github.com/PaddlePaddle/PaddleTest/pull/2877)  |
| test_functional_conv1d_transpose.py | paddle.nn.functional.conv1d_transpose 是否在PIR下需要适配 |  ✅https://github.com/PaddlePaddle/Paddle/pull/65754 |
| test_TransformerDecoderLayer.py | 动态图下`paddle.framework.random._manual_program_seed`使用OldIrGuard进行包裹 | ✅[PaddleTest#2878](https://github.com/PaddlePaddle/PaddleTest/pull/2878)  |
| test_functional_transpose_conv1d.py | 同 conv1d_transpose API 适配| ✅https://github.com/PaddlePaddle/Paddle/pull/65754  |
| test_transpose_conv1d.py | 同 conv1d_transpose API 适配 |✅https://github.com/PaddlePaddle/Paddle/pull/65754   |
| test_functional_utils_weight_norm.py | 涉及 paddle.nn.utils.weight_norm(layer, dim=dim) |  ✅https://github.com/PaddlePaddle/Paddle/pull/66017  |
| test_functional_transpose_conv1d_float64.py | 同 conv1d_transpose API 适配 |✅https://github.com/PaddlePaddle/Paddle/pull/65754   |
| test_functional_pad.py | 同 test_squeeze.py | ✅[PaddleTest#2883](https://github.com/PaddlePaddle/PaddleTest/pull/2883)  |
| test_transpose_conv1d_float64.py | 同 conv1d_transpose API 适配 | ✅https://github.com/PaddlePaddle/Paddle/pull/65754  |
| test_spectral_norm.py | 精度不一样? | ✅[PaddleTest#2885](https://github.com/PaddlePaddle/PaddleTest/pull/2885)  |
| test_beamsearchdecoder.py | 异常信息与旧IR不同 `(InvalidType) full(): argument (position 2) must be double, but got str (at ..\paddle\fluid\pybind\op_function_common.cc:384)` | ✅[PaddleTest#2879](https://github.com/PaddlePaddle/PaddleTest/pull/2879)  |
| test_Transformer.py | 属于tensor.shape不一致 `[]` `()` | ✅[PaddleTest#2877](https://github.com/PaddlePaddle/PaddleTest/pull/2877)  |
| test_functional_transpose_conv2d.py | 同 test_squeeze.py |✅[PaddleTest#2883](https://github.com/PaddlePaddle/PaddleTest/pull/2883)   |

## Others

1. 对于一个单测文件，不细分的测试。可能会出现 enable_static 的泄露导致看着的失败案例比实际多

test_beamsearchdecoder.py 中 `test_beamsearchdecoder8` 测试失败异常退出，未执行末尾的 paddle.disable_static()，文件单测时，给下面未开启静态图的 `test_beamsearchdecoder9` 进行 PIR 测试。
